agent/acs/handler/task_manifest_common.go

```diff
@@ -0,0 +1,23 @@
+package handler
+
+import (
+	"sync"
+)
+
+type manifestMessageIDAccessor struct {
+	messageID string
+	lock      sync.RWMutex
```

Comment:
[Not required in this PR] I wonder if there's a way to define thread-safe variables at a higher level in Go. Having to manage a lock manually seems so archaic. 😅

Reply:
Hmm yeah, off the top of my head using a `sync.RWMutex` here to protect from concurrent access seems to be the most direct/straightforward approach here (as opposed to other common thread safety mechanisms from `sync` in Go). But I can take an action to look into this further.

Reply:
I think this is the most straightforward way in Go. I was comparing Go to other languages. For example, Java has atomic variables and Haskell has MVars.